### PR TITLE
[AIRFLOW-6130] Make Cassandra to GCS operator pylint compatible

### PR DESCRIPTION
**Pull request by turbaszek**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6130
### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/operators/cassandra_to_gcs.py [hunk 1]
```diff
@@ -141,8 +140,10 @@ def __init__(self,
         'VarcharType': 'STRING',
     }
 
-    def execute(self, context):
-        cursor = self._query_cassandra()
+    def execute(self, context: Dict):
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def execute(self, context: Dict[str, str]):
```

File: airflow/operators/cassandra_to_gcs.py [hunk 2]
```diff
@@ -209,29 +201,38 @@ def _write_local_schema_file(self, cursor):
...
-            schema.append(self.generate_schema_dict(name, type))
+        for name, type_ in zip(cursor.column_names, cursor.column_types):
+            schema.append(self.generate_schema_dict(name, type_))
         json_serialized_schema = json.dumps(schema).encode('utf-8')
 
         tmp_schema_file_handle.write(json_serialized_schema)
         return {self.schema_filename: tmp_schema_file_handle}
 
-    def _upload_to_gcs(self, files_to_upload):
+    def _upload_to_gcs(self, files_to_upload: Dict):
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def _upload_to_gcs(self, files_to_upload: Dict[str, str]):
```

File: airflow/operators/cassandra_to_gcs.py [hunk 3]
```diff
@@ -209,29 +201,38 @@ def _write_local_schema_file(self, cursor):
...
+                gzip=self.gzip
+            )
 
     @classmethod
-    def generate_data_dict(cls, names, values):
-        row_dict = {}
-        for name, value in zip(names, values):
-            row_dict.update({name: cls.convert_value(name, value)})
-        return row_dict
+    def generate_data_dict(cls, names: Iterable[str], values: Any) -> Dict:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def generate_data_dict(cls, names: Iterable[str], values: Any) -> Dict[str, Any]:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 4]
```diff
@@ -209,29 +201,38 @@ def _write_local_schema_file(self, cursor):
...
+        """
+        Generates data structure that will be stored as file in GCS.
+        """
+        return {n: cls.convert_value(v) for n, v in zip(names, values)}
 
     @classmethod
-    def convert_value(cls, name, value):
+    def convert_value(cls, value):  # pylint:disable=too-many-return-statements
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def convert_value(cls, value: Optional[Any]) -> Optional[Any]:  # pylint:disable=too-many-return-statements
```

File: airflow/operators/cassandra_to_gcs.py [hunk 5]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
-            return cls.convert_map_type(name, value)
+            return cls.convert_map_type(value)
         else:
-            raise AirflowException('unexpected value: ' + str(value))
+            raise AirflowException('Unexpected value: ' + str(value))
 
     @classmethod
-    def convert_array_types(cls, name, value):
-        return [cls.convert_value(name, nested_value) for nested_value in value]
+    def convert_array_types(cls, value: Union[list, SortedSet]) -> List:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def convert_array_types(cls, value: Union[List[Any], SortedSet[Any]]) -> List[Any]:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 6]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
+        """
+        Maps convert_value over array.
+        """
+        return [cls.convert_value(nested_value) for nested_value in value]
 
     @classmethod
-    def convert_user_type(cls, name, value):
+    def convert_user_type(cls, value):
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def convert_user_type(cls, value: Any) -> Dict[str, Any]:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 7]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
         index is determined by the order of the tuple elements defined in cassandra.
         """
-        names = ['field_' + str(i) for i in range(len(value))]
-        values = [cls.convert_value(name, value) for name, value in zip(names, value)]
+        names = ['field_' + str(i) for i in range(len(values))]
         return cls.generate_data_dict(names, values)
 
     @classmethod
-    def convert_map_type(cls, name, value):
+    def convert_map_type(cls, value: OrderedMapSerializedKey) -> List[Dict]:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def convert_map_type(cls, value: OrderedMapSerializedKey) -> List[Dict[Any, Any]]:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 8]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
-                'value': cls.convert_value('value', v)
+                'key': cls.convert_value(k),
+                'value': cls.convert_value(v)
             })
         return converted_map
 
     @classmethod
-    def generate_schema_dict(cls, name, type):
-        field_schema = dict()
+    def generate_schema_dict(cls, name: str, type_) -> Dict:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def generate_schema_dict(cls, name: str, type_: Any) -> Dict[str, Any]:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 9]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
-            for name, type in zip(names, types):
-                field = cls.generate_schema_dict(name, type)
-                fields.append(field)
-
-        return fields
-
-    @classmethod
-    def is_simple_type(cls, type):
-        return type.cassname in CassandraToGoogleCloudStorageOperator.CQL_TYPE_MAP
+    def get_bq_fields(cls, type_) -> List[Dict]:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def get_bq_fields(cls, type_: Any) -> List[Dict[str, Any]]:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 10]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
+
+        if types and not names and type_.cassname == 'TupleType':
+            names = ['field_' + str(i) for i in range(len(types))]
+        elif types and not names and type_.cassname == 'MapType':
+            names = ['key', 'value']
+
+        return [cls.generate_schema_dict(n, t) for n, t in zip(names, types)]
+
+    @staticmethod
+    def is_simple_type(type_) -> bool:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def is_simple_type(type_: Any) -> bool:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 11]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
+        """
+        Check if type is a simple type.
+        """
+        return type_.cassname in CassandraToGoogleCloudStorageOperator.CQL_TYPE_MAP
 
-    @classmethod
-    def is_array_type(cls, type):
-        return type.cassname in ['ListType', 'SetType']
+    @staticmethod
+    def is_array_type(type_) -> bool:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def is_array_type(type_: Any) -> bool:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 12]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
+        """
+        Check if type is an array type.
+        """
+        return type_.cassname in ['ListType', 'SetType']
 
-    @classmethod
-    def is_record_type(cls, type):
-        return type.cassname in ['UserType', 'TupleType', 'MapType']
+    @staticmethod
+    def is_record_type(type_) -> bool:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def is_record_type(type_: Any) -> bool:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 13]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
+        Checks the record type.
+        """
+        return type_.cassname in ['UserType', 'TupleType', 'MapType']
 
     @classmethod
-    def get_bq_type(cls, type):
-        if cls.is_simple_type(type):
-            return CassandraToGoogleCloudStorageOperator.CQL_TYPE_MAP[type.cassname]
-        elif cls.is_record_type(type):
+    def get_bq_type(cls, type_) -> str:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def get_bq_type(cls, type_: Any) -> str:
```

File: airflow/operators/cassandra_to_gcs.py [hunk 14]
```diff
@@ -247,120 +248,140 @@ def convert_value(cls, name, value):
...
+        elif cls.is_array_type(type_):
+            return cls.get_bq_type(type_.subtypes[0])
         else:
-            raise AirflowException('Not a supported type: ' + type.cassname)
+            raise AirflowException('Not a supported type_: ' + type_.cassname)
 
     @classmethod
-    def get_bq_mode(cls, type):
-        if cls.is_array_type(type) or type.cassname == 'MapType':
+    def get_bq_mode(cls, type_) -> str:
```
Comment (potiuk) [inline airflow/operators/cassandra_to_gcs.py]:
```suggestion
    def get_bq_mode(cls, type_: Any) -> str:
```

**Review by potiuk — CHANGES_REQUESTED** [review]
* First of all - there are few places where key of the Dict is limited to str.
* Secondly I think there is a big value in using Any, even if it can be skipped. First of all you signal to the developer that comes after you that you deliberately accept Any type passed (and it's not an overlook). Also you can use Optional[Any] and detect the cases where None values can cause problems.

File: airflow/operators/cassandra_to_gcs.py (hunk 2 above)
Comment (turbaszek) [reply airflow/operators/cassandra_to_gcs.py]: Are you sure about that?

Comment (potiuk) [reply airflow/operators/cassandra_to_gcs.py]: Not sure. But MyPy should be able to check it if we specify it here.

**Review by potiuk — APPROVED** [review]

**Pull request merged**